cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -1604,6 +1634,17 @@ impl RegMemImm {
             RegMemImm::Imm { .. } => {}
         }
     }
+
+    fn map_src_reg_as_def<RUM: RegUsageMapper>(&mut self, mapper: &RUM) {
```

Comment:
Just as a sanity check .. do we need special-case handling for `map_regs`?  IIUC, if there is no special-case handling then the mapping for the `reg` component of the insn will assert, complaining that the `def` and `use` maps are inconsistent .. yes?  Can you confirm?

Reply:
I didn't run into any issues with this, and i could confirm it was probably used. This patch does adjust both reg usage collection and reg usage mapping, so nothing should be missing as far as i can tell.